builder/azure/arm/config.go

```diff
@@ -215,7 +217,7 @@ type Config struct {
 	// The number of replicas of the Image Version to be created per region. This
 	// property would take effect for a region when regionalReplicaCount is not specified.
 	// Replica count must be between 1 and 100, but 50 replicas should be sufficient for most use cases.
-	SharedGalleryImageVersionReplicaCount int32 `mapstructure:"shared_image_gallery_replica_count" required:"false"`
+	SharedGalleryImageVersionReplicaCount int64 `mapstructure:"shared_image_gallery_replica_count" required:"false"`
```

Comment:
In the new SDK this value is an int64, because this was being parsed as an int32, stored in the state bag, and then in `step_publish_to_shared_image_gallery` it was retrieved as an int64, this int would be incorrectly registered, leading to https://github.com/hashicorp/packer-plugin-azure/issues/341